src/WebJobs.Script/OutOfProc/Http/Configuration/HttpInvokerOptionsSetup.cs

```diff
@@ -42,8 +44,19 @@ public void Configure(HttpInvokerOptions options)
                 };
 
                 options.Arguments.ExecutableArguments.AddRange(options.Description.Arguments);
+                options.Port = GetUnusedTcpPort();
                 _logger.LogDebug("Configured httpInvoker with DefaultExecutalbePath: {exepath} with arguments {args}", options.Description.DefaultExecutablePath, options.Arguments);
             }
         }
+
+        internal static int GetUnusedTcpPort()
+        {
+            // TODO verify in Azure
+            TcpListener tcpListener = new TcpListener(IPAddress.Loopback, 0);
+            tcpListener.Start();
+            int port = ((IPEndPoint)tcpListener.LocalEndpoint).Port;
+            tcpListener.Stop();
+            return port;
+        }
```

Comment:
If a port is found here, and then the listener is not bound to it anymore, there is a chance (miniscule?) that the port number found here could be bound to something else before it goes to the _processFactory?

Reply:
true! but host as of now does not open any other ports so should be ok

Reply:
The OS could still assign that port to something else meanwhile, right?

Reply:
yes. possible. As of now, I could not find a better way to do this.